ts/src/binance.ts

```diff
@@ -13145,6 +13146,47 @@ export default class binance extends Exchange {
         return this.parseGreeks (response[0], market);
     }
 
+    /**
+     * @method
+     * @name binance#fetchAllGreeks
+     * @description fetches all option contracts greeks, financial metrics used to measure the factors that affect the price of an options contract
+     * @see https://developers.binance.com/docs/derivatives/option/market-data/Option-Mark-Price
+     * @param {string[]} [symbols] unified symbols of the markets to fetch greeks for, all markets are returned if not assigned
+     * @param {object} [params] extra parameters specific to the exchange API endpoint
+     * @returns {object} a [greeks structure]{@link https://docs.ccxt.com/#/?id=greeks-structure}
+     */
+    async fetchAllGreeks (symbols: Strings = undefined, params = {}): Promise<Greeks[]> {
+        await this.loadMarkets ();
+        symbols = this.marketSymbols (symbols, undefined, true, true, true);
+        const request: Dict = {};
+        let market = undefined;
+        if (symbols !== undefined) {
+            if (symbols.length === 1) {
```

Comment:
@Dan-krm 
![image](https://github.com/user-attachments/assets/934e0c66-be17-46cf-a4d4-2bbf258e4f9a)

.length like that will be translated to strlen in php